### ISSUE TITLE
orm.PrimaryKey and orm.Set need explicit annotations

### ISSUE THREAD

**Issue by fluffy-critter** [Jonesus/pony-stubs#3]
When I have a model like:

```
class Foo(orm.core.Entity):
    name = orm.PrimaryKey(str)
    bar = orm.Set("Bar")

class Bar(orm.core.Entity):
    foo = orm.Set(Foo)
```

mypy generates errors such as `need type annotation for "name"` and `need type annotation for "bar"`. I'm able to work around this by adding annotations such as:

```
class Foo(orm.core.Entity):
    name:orm.PrimaryKey[str] = orm.PrimaryKey(str)
```

and so on.

**Comment by fluffy-critter** [comment]
Oh, and those annotations only make mypy happy; Python itself ends up dying with errors like:

```
  File "/Users/fluffy/projects/Publ/publ/model.py", line 24, in GlobalConfig
    key: orm.PrimaryKey[str] = orm.PrimaryKey(str)
TypeError: 'type' object is not subscriptable
```

which makes sense in retrospect.

**Comment by Jonesus** [comment]
What version of Python are you running? Adding 
```python
from __future__ import annotations
```
on top of your `model.py` should help, but IIRC only on Python >=3.7

**Comment by Jonesus** [comment]
As for the original issue (got carried away :sweat_smile: ) when declaring
```python
bar = orm.Set("Bar")
```
We can't infer the string `"Bar"` to mean the class `Bar` (as we can in `foo = orm.Set(Foo)`), using `orm.Set()` with a string argument will require a manual annotation for as long as we try to do this without any MyPy plugins.

The `PrimaryKey` should be inferable, though, so I'll try to figure it out!

**Comment by fluffy-critter** [comment]
I'm targeting Python >= 3.7, and currently running on 3.8.

When I add that import to the top of `model.py` I still get the errors requiring a type annotation for `orm.PrimaryKey`, which doesn't seem correct.

Also in the README it isn't clear where `Set` is being imported from in the examples e.g.

```
class Customer(DbEntity):
    ...
    orders: Set["Order"] = Set("Order")
```

It makes sense that forward declarations can't be done automatically but I'm not sure how to actually do the declaration here to keep both mypy and python itself happy.

**Comment by Jonesus** [comment]
> When I add that import to the top of model.py I still get the errors requiring a type annotation for orm.PrimaryKey, which doesn't seem correct.

You're totally right, and I think I managed to fix it now, can you try updating to `0.5.1`?

>Also in the README it isn't clear where Set is being imported from in the examples e.g.

Good catch, thank you :pray:  Were the examples otherwise good enough, or is there something that could be improved?

> but I'm not sure how to actually do the declaration here to keep both mypy and python itself happy.

did the `from __future__ import annotations` stop python itself from crashing though? I can't seem to replicate the crashes with the import myself if that's not the case

**Comment by fluffy-critter** [comment]
> > When I add that import to the top of model.py I still get the errors requiring a type annotation for orm.PrimaryKey, which doesn't seem correct.
> 
> You're totally right, and I think I managed to fix it now, can you try updating to `0.5.1`?

0.5.1 seems to have fixed it, thanks!

> > Also in the README it isn't clear where Set is being imported from in the examples e.g.
> 
> Good catch, thank you 🙏 Were the examples otherwise good enough, or is there something that could be improved?

I tried using the 

```
class BaseEntity(db.Entity): # type:ignore
    pass
```

approach and that failed with

```
    if primary_keys: throw(ERDiagramError, 'Primary key cannot be redefined in derived classes')
```

For now I've gone back to:

```
db = orm.Database()  # pylint:ignore=invalid-name
DbEntity: TypeAlias = db.Entity  # type:ignore
```

which seems to be working.

> > but I'm not sure how to actually do the declaration here to keep both mypy and python itself happy.
> 
> did the `from __future__ import annotations` stop python itself from crashing though? I can't seem to replicate the crashes with the import myself if that's not the case

The `from __future__ import annotations` appears to have stopped Python from crashing and allows `foo:orm.Set[Bar] = orm.Set("bar")` to work correctly, yes.

**Comment by fluffy-critter** [comment]
Hmm, something in `mypy` now seems to be going into an infinite loop, or at least it's taking a rather long time to finish. Unfortunately interrupting mypy just gives an `Interrupted` error rather than a stack trace so I'm not sure where it's hanging.

**Comment by fluffy-critter** [comment]
Okay it did finish eventually. Running it with `-v` indicates that there's just a lot of non-Pony-related things that are causing a lot of processing to occur. Weird.

**Comment by Jonesus** [comment]
Yeah it also seems that the `class BaseEntity` thing was really naively thought out and didn't work at all after all :sweat_smile: I'll remove that from the readme
